src/fork.rs

```diff
@@ -125,11 +125,14 @@ impl<S: ForkSource> ForkStorage<S> {
         fork: Option<ForkDetails>,
         system_contracts_options: &system_contracts::Options,
         use_evm_emulator: bool,
+        override_chain_id: Option<u32>,
     ) -> Self {
         let chain_id = fork
             .as_ref()
             .and_then(|d| d.overwrite_chain_id)
-            .unwrap_or(L2ChainId::from(TEST_NODE_NETWORK_ID));
+            .unwrap_or(L2ChainId::from(
+                override_chain_id.unwrap_or(TEST_NODE_NETWORK_ID),
```

Comment:
We'll also want to update `src/node/net.rs` as that's using the constant `TEST_NODE_NETWORK_ID` for `net_version` RPC requests

Reply:
done - PTAL